server/src/test/java/org/elasticsearch/index/search/nested/NestedSortingTests.java

```diff
@@ -99,7 +99,7 @@ public void testDuel() throws Exception {
         MultiValueMode sortMode = randomFrom(Arrays.asList(MultiValueMode.MIN, MultiValueMode.MAX));
         DirectoryReader reader = DirectoryReader.open(writer);
         reader = ElasticsearchDirectoryReader.wrap(reader, new ShardId(indexService.index(), 0));
-        IndexSearcher searcher = newSearcher(reader);
+        IndexSearcher searcher = new IndexSearcher(reader);
```

Comment:
Dear reviewer, this fails randomly due to the reader getting randomly wrapped internally by `newSearcher`. We expect this type of reader and it cannot be wrapped. 

```
org.elasticsearch.index.search.nested.NestedSortingTests > testMultiLevelNestedSorting {seed=[4A3E25BDAA026372:B462EF24EC743438]} FAILED
    java.lang.IllegalStateException: Null shardId. If you got here from a test, you need to wrap the directory reader. see for example AggregatorTestCase#wrapInMockESDirectoryReader.  If you got here in production, please file a bug.
        at __randomizedtesting.SeedInfo.seed([4A3E25BDAA026372:B462EF24EC743438]:0)
        at org.elasticsearch.index.cache.bitset.BitsetFilterCache.getAndLoadIfNotPresent(BitsetFilterCache.java:149)
        at org.elasticsearch.index.cache.bitset.BitsetFilterCache$QueryWrapperBitSetProducer.getBitSet(BitsetFilterCache.java:224)
        at org.elasticsearch.index.fielddata.IndexFieldData$XFieldComparatorSource$Nested.rootDocs(IndexFieldData.java:141)
        at org.elasticsearch.index.fielddata.fieldcomparator.LongValuesComparatorSource.getNumericDocValues(LongValuesComparatorSource.java:90)
        at org.elasticsearch.index.fielddata.fieldcomparator.LongValuesComparatorSource$1$1.getNumericDocValues(LongValuesComparatorSource.java:109)
        at org.apache.lucene.search.comparators.NumericComparator$NumericLeafComparator.<init>(NumericComparator.java:106)
        at org.apache.lucene.search.comparators.LongComparator$LongLeafComparator.<init>(LongComparator.java:65)
        at org.elasticsearch.index.fielddata.fieldcomparator.LongValuesComparatorSource$1$1.<init>(LongValuesComparatorSource.java:106)
        at org.elasticsearch.index.fielddata.fieldcomparator.LongValuesComparatorSource$1.getLeafComparator(LongValuesComparatorSource.java:106)
        at org.apache.lucene.search.FieldValueHitQueue.getComparators(FieldValueHitQueue.java:176)
        at org.apache.lucene.search.TopFieldCollector$TopFieldLeafCollector.<init>(TopFieldCollector.java:64)
        at org.apache.lucene.search.TopFieldCollector$SimpleFieldCollector$1.<init>(TopFieldCollector.java:198)
        at org.apache.lucene.search.TopFieldCollector$SimpleFieldCollector.getLeafCollector(TopFieldCollector.java:198)
        at org.apache.lucene.search.FilterCollector.getLeafCollector(FilterCollector.java:38)
        at org.apache.lucene.tests.search.AssertingCollector.getLeafCollector(AssertingCollector.java:52)
        at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:762)
        at org.apache.lucene.tests.search.AssertingIndexSearcher.search(AssertingIndexSearcher.java:78)
        at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:694)
        at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:688)
        at org.apache.lucene.search.IndexSearcher.searchAfter(IndexSearcher.java:668)
        at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:585)
        at org.elasticsearch.index.search.nested.NestedSortingTests.search(NestedSortingTests.java:823)
        at org.elasticsearch.index.search.nested.NestedSortingTests.testMultiLevelNestedSorting(NestedSortingTests.java:621)
```